### ISSUE TITLE
configmap/secret - default takes precedence

### ISSUE THREAD

**Issue by filojalo** [apache/camel-karavan#1453]
### Describe the bug

When using configmap/secret property placeholder syntax
{{configmap:secondmap/test:default}} it will choose default even if key in configmap is present.

### Steps to reproduce the behavior

Using a simple testing camel route to test configMap and secret behaviour
```
- rest:
    id: rest-d471
    post:
      - id: post-126a
        path: /injectionTest
        to: direct:injectionTest
- route:
    id: route-a292
    from:
      id: from-b12a
      uri: direct
      parameters:
        name: injectionTest
      steps:
        - to:
            id: to-bab9
            uri: kubernetes-config-maps
            parameters:
              namespace: default
              masterUrl: kubernetes.default.svc:443
              operation: listConfigMaps
        - marshal:
            id: marshal-55e5
            json:
              id: json-b0d1
        - log:
            id: log-d04e
            message: >-
              Printing test configmap without default
              {{configmap:secondmap/test}}
        - log:
            id: log-3396
            message: Printing test secret without default {{secret:secondsecret/test}}
        - log:
            id: log-f123
            message: >-
              Printing test configmap with default
              {{configmap:secondmap/test:default}}
        - log:
            id: log-f123
            message: >-
              Printing test secret with default
              {{configmap:secondsecret/test:default}}
```

Then running the built integration in my minikube testing cluster it will printout the actual value of key test if default is not used. But if default is used it will printout the default.
![image](https://github.com/user-attachments/assets/aac0caf6-8fb5-464c-8423-4a18cf0aaea6)

More likely a Camel issue? Running on Karavan/Camel 4.7.0 as 4.8.0 has this bug https://issues.apache.org/jira/browse/CAMEL-21316

### Variant

Web Application

### Container Management (if applicable)

Docker

### Operating System (if applicable)

Windows

### Version

4.7.0

**Comment by oscerd** [comment]
This seems to be a problem in Camel Kubernetes configmap properties source.

**Comment by filojalo** [comment]
Created https://issues.apache.org/jira/browse/CAMEL-21446

**Comment by davsclaus** [comment]
fixed in camel 4.8.2
